app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -56,4 +64,9 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
         startActivity(intent)
     }
 
+    override fun setDepartureTable(departures: List<DepartureInformation>) {
+        (recyclerView.adapter as RecyclerAdapter).setDepartures(departures)
+        (recyclerView.adapter as RecyclerAdapter).notifyDataSetChanged()
```

Comment:
I think we can delete this line and just call notifyDataSetChanged() at the end of the setDepartures() function